ext/opcache/ZendAccelerator.c

```diff
@@ -505,39 +489,35 @@ zend_string *accel_new_interned_string(zend_string *str)
 	HT_HASH(&ZCSG(interned_strings), nIndex) = HT_IDX_TO_HASH(idx);
 	zend_string_release(str);
 	return p->key;
-#else
-	return str;
-#endif
 }
 
-#ifndef ZTS
 /* Copy PHP interned strings from PHP process memory into the shared memory */
-static void accel_use_shm_interned_strings(void)
+static void accel_copy_permanent_strings(zend_new_interned_string_func_t new_interned_string)
```

Comment:
On ASLR enabled systems, like Windows by default or hardened Linux, pointers to interned strings are likely differ. So in first place, by copying we ensure the pointers are same in all the processes. If something is missed here, the interning will happen on demand in zend_persist.c and zend_persist_calc.c parts on demand. See ADD_INTERNED_STRING, zend_accel_store_interned_string and so on there.

It might be, that on some systems the pointers to the local interned strings could stay same, But as we can't guarantee it, it's played safe. While it were possible to add a configure option and modify the code for not saving to SHM, i'd not go for that, as it is really not a reliable behavior.